src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/DocumentResource.kt

```diff
@@ -0,0 +1,63 @@
+package uk.gov.justice.digital.hmpps.nomisprisonerapi.core
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.media.Content
+import io.swagger.v3.oas.annotations.media.Schema
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import org.springframework.http.MediaType
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.validation.annotation.Validated
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.config.ErrorResponse
+
+@RestController
+@Validated
+@RequestMapping(produces = [MediaType.APPLICATION_JSON_VALUE])
+class DocumentResource(private val documentService: DocumentService) {
+  @PreAuthorize("hasRole('ROLE_NOMIS_DOCUMENTS')")
+  @GetMapping("/documents/{id}", produces = ["application/x-zip-compressed"])
+  @Operation(
+    summary = "Retrieve a document",
+    description = "Retrieve a document by its id. Requires role NOMIS_CSIP",
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "OK",
+      ),
+      ApiResponse(
+        responseCode = "400",
+        description = "Bad request",
+        content = [
+          Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class)),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [
+          Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class)),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Forbidden, requires role NOMIS_DOCUMENTS",
+        content = [
+          Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class)),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "Document not found",
+        content = [
+          Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class)),
+        ],
+      ),
+    ],
+  )
+  fun getDocument(
+    @Schema(description = "The document id") @PathVariable id: Long,
+  ): ByteArray? = documentService.getDocumentById(id)
+}
```

Comment:
still to add - get all document ids/summaries by matching template name and bookingId

Reply:
any reason why this returns an optional ByteArray?

Reply:
no - it should return a ByteArray - I changed the call at the last minute to throw an exception instead